scripts/autogen

```diff
@@ -1244,6 +1244,251 @@ class BibliographyEntry:
         return authors
 
 
+def update_via_copy(infile_full, outfile_full, dry_run=False, transform=None):
+    status_update("copy", f"{infile_full} -> {outfile_full}")
+
+    with open(infile_full, "r") as f:
+        content = f.read()
+
+    if transform is not None:
+        content = transform(content)
+
+    update_file(outfile_full, content, dry_run=dry_run)
+
+
+def update_via_remove(filename, dry_run=False):
+    if dry_run is True:
+        print(
+            f"Autogenerated file {filename} needs removing. Have you called scripts/autogen?",
+            file=sys.stderr,
+        )
+        exit(1)
+
+    # Remove the file
+    os.remove(filename)
```

Comment:
Can we stick to the same order of functions as in mlkem-native, to minimize the diff. 
In mlkem-native `update_via_copy` and `update_via_remove` come much later in the file.